adapters/mgidX/mgidX.go

```diff
@@ -87,7 +95,44 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.E
 	return adapterRequests, nil
 }
 
+func (a *adapter) getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ImpExtMgidX, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, fmt.Errorf("Bidder extension not provided or can't be unmarshalled")
+	}
+
+	var mgidXExt openrtb_ext.ImpExtMgidX
+	if err := json.Unmarshal(bidderExt.Bidder, &mgidXExt); err != nil {
+		return nil, fmt.Errorf("Error while unmarshaling bidder extension")
+	}
+
+	return &mgidXExt, nil
+}
+
+func (a *adapter) buildEndpointURL(params *openrtb_ext.ImpExtMgidX) (string, error) {
+	var endpointParams macros.EndpointTemplateParams
+
+	if params.Region == "eu" {
+		endpointParams = macros.EndpointTemplateParams{Host: "eu"}
+	} else {
+		endpointParams = macros.EndpointTemplateParams{Host: "us-east-x"}
```

Comment:
This implementation will prevent a host company from sending all traffic from their EU data centers to yours, as the control is entirely placed on the publisher. Would you consider falling back to a host specified user endpoint instead of hardcoding US East?

Reply:
Actually, a host could still override the endpoint to include the eu subdomain and omit the macro. I think that's fine, if you could please make that clear through documentation.